server/src/main/java/org/elasticsearch/index/fielddata/IndexFieldDataService.java

```diff
@@ -74,11 +72,10 @@ public void onRemoval(ShardId shardId, String fieldName, boolean wasEvicted, lon
 
 
     public IndexFieldDataService(IndexSettings indexSettings, IndicesFieldDataCache indicesFieldDataCache,
-                                 CircuitBreakerService circuitBreakerService, MapperService mapperService) {
+                                 CircuitBreakerService circuitBreakerService) {
         super(indexSettings);
         this.indicesFieldDataCache = indicesFieldDataCache;
         this.circuitBreakerService = circuitBreakerService;
-        this.mapperService = mapperService;
```

Comment:
good catch! I think this one could go in separately. I may open a PR shortly around it.